### PR TITLE
Feat ppo

### PR DESCRIPTION
**Pull request by AGKhalil**
that's right. another one for the books

### CODE REVIEW

**Comment by AGKhalil** [comment]
Im keeping a2c and ppo on the same branch cause a lot of the new changes relate to both. otherwise there will be overlap between the two pull requests.

**Comment by MihaiAnca13** [comment]
I just had a very quick scan through and noticed you have different networks for the actor and critic. In A2C, I believe they must share the same base, but have different heads.

**Comment by MihaiAnca13** [comment]
Also, the optimizers must be trained separately. The training step is being called twice per batch because you have two optimizers. This means that with the current code, you are passing the same loss to both actor and critic. It might be that this is a different way of doing it, so please let me know if I'm wrong.

**Comment by AGKhalil** [comment]
I’ll double check the different networks thing. If it is I’ll modify it. PPO is using two different nets, but i’ve seen implementations with only one net and two heads. 

Regarding the two iterations per training step. This has been happening before the two optimizers were added in. With the new pytorch lightning version we should be able to specify the number of training steps to run. Essentially, after this pull request i was gonna fix the double iterations thing. Though i don’t think it’s related to the optimizers.

**Comment by AGKhalil** [comment]
I see nothing saying we need to use only one network for A2C. If you have no other issues @MihaiAnca13 , I'll merge.

I'll deal with the multiple iterations per step thing in another branch.

**Comment by MihaiAnca13** [comment]
If you've trained it and it works, I have no further comments.
